src/api/pact-pair.js

```diff
@@ -63,67 +63,194 @@ export const getPairList = async () => {
+        return accum;
+      }, '');
+    const tokenPairListWithoutBooster = Object.values(pairTokens)
+      .filter((token) => !token.isBoosted)
+      .reduce((accum, pair) => {
+        accum += `[${pair.name.split(':').join(' ')}] `;
+        return accum;
+      }, '');
+
+    const boosterResult = await dataWithBooster(account, tokenPairListWithBooster);
+
+    const noBoosterResult = await dataWithoutBooster(account, tokenPairListWithoutBooster);
+
+    const totalResult = boosterResult.concat(noBoosterResult);
+
+    return totalResult;
+  } catch (e) {
+    return handleError(e);
+  }
+};
+
+export const getOneSideLiquidityPairInfo = async (amountA, slippage, token0, token1) => {
+  try {
     let data = await pactFetchLocal(
       `
-            (namespace 'free)
-
-            (module ${KADDEX_NAMESPACE}-read G
-
-              (defcap G ()
-                true)
-
-              (defun pair-info (pairList:list)
-                (let* (
-                  (token0 (at 0 pairList))
-                  (token1 (at 1 pairList))
-                  (p (${KADDEX_NAMESPACE}.exchange.get-pair token0 token1))
-                  (reserveA (${KADDEX_NAMESPACE}.exchange.reserve-for p token0))
-                  (reserveB (${KADDEX_NAMESPACE}.exchange.reserve-for p token1))
-                  (totalBal (${KADDEX_NAMESPACE}.tokens.total-supply (${KADDEX_NAMESPACE}.exchange.get-pair-key token0 token1)))
-                  (acctBal
-                      (try 0.0 (${KADDEX_NAMESPACE}.tokens.get-balance (${KADDEX_NAMESPACE}.exchange.get-pair-key token0 token1) ${JSON.stringify(
-        account
-      )})
-                    ))
-                )
-                [(${KADDEX_NAMESPACE}.exchange.get-pair-key token0 token1)
-                 reserveA
-                 reserveB
-                 totalBal
-                 acctBal
-                 (* reserveA (/ acctBal totalBal))
-                 (* reserveB (/ acctBal totalBal))
-               ]
-              ))
-            )
-            (map (${KADDEX_NAMESPACE}-read.pair-info) [${tokenPairList}])
-             `
+    (namespace 'free)
+
+    (module ${KADDEX_NAMESPACE}-read G
+
+      (defcap G ()
+        true)
+
+      (defun pair-info ()
+        (let* (
+          (p (${KADDEX_NAMESPACE}.exchange.get-pair ${token0} ${token1}))
+          (pair-account (at 'account p))
+          (amountB (${KADDEX_NAMESPACE}.wrapper.get-other-side-token-amount-after-swap ${amountA} ${token0} ${token1} (+ ${amountA} ${slippage})))
+          (reserveA (${KADDEX_NAMESPACE}.exchange.reserve-for p ${token0}))
+          (amountA-after-swap (- ${amountA} (${KADDEX_NAMESPACE}.wrapper.get-one-sided-liquidity-swap-amount reserveA ${amountA})))
+          (amountA-min (${KADDEX_NAMESPACE}.exchange.truncate ${token0} (* amountA-after-swap (- ${amountA} ${slippage}))))
+          (amountB-min (${KADDEX_NAMESPACE}.exchange.truncate ${token1} (* (/ amountB (+ ${amountA} ${slippage})) (- ${amountA} ${slippage}))))
```

Comment:
The parts of the code that uses the slippage should be `(+ 1.0 ${slippage})` and `(- 1.0 ${slippage})` instead of using `${amountA}`. This is assuming that a slippage of 5% is represented as `0.05`
```suggestion
          (p (${KADDEX_NAMESPACE}.exchange.get-pair ${token0} ${token1}))
          (pair-account (at 'account p))
          (amountB (${KADDEX_NAMESPACE}.wrapper.get-other-side-token-amount-after-swap ${amountA} ${token0} ${token1} (+ 1.0 ${slippage})))
          (reserveA (${KADDEX_NAMESPACE}.exchange.reserve-for p ${token0}))
          (amountA-after-swap (- ${amountA} (${KADDEX_NAMESPACE}.wrapper.get-one-sided-liquidity-swap-amount reserveA ${amountA})))
          (amountA-min (${KADDEX_NAMESPACE}.exchange.truncate ${token0} (* amountA-after-swap (- 1.0 ${slippage}))))
          (amountB-min (${KADDEX_NAMESPACE}.exchange.truncate ${token1} (* (/ amountB (+ 1.0 ${slippage})) (- 1.0 ${slippage}))))
```

(I only really checked the pact code snippets btw)